homeassistant/components/influxdb/sensor.py

```diff
@@ -121,63 +142,30 @@ def validate_query_format_for_version(conf: Dict) -> Dict:
 )
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+def setup_platform(hass, conf, add_entities, discovery_info=None):
     """Set up the InfluxDB component."""
-    use_v2_api = config[CONF_API_VERSION] == API_VERSION_2
-    queries = None
-
-    if use_v2_api:
-        influx_conf = {
-            "url": config[CONF_URL],
-            "token": config[CONF_TOKEN],
-            "org": config[CONF_ORG],
-        }
-        bucket = config[CONF_BUCKET]
-        queries = config[CONF_QUERIES_FLUX]
-
-        for v2_query in queries:
-            if CONF_BUCKET not in v2_query:
-                v2_query[CONF_BUCKET] = bucket
-
-    else:
-        influx_conf = {
-            "database": config[CONF_DB_NAME],
-            "verify_ssl": config[CONF_VERIFY_SSL],
-        }
-
-        if CONF_USERNAME in config:
-            influx_conf["username"] = config[CONF_USERNAME]
-
-        if CONF_PASSWORD in config:
-            influx_conf["password"] = config[CONF_PASSWORD]
-
-        if CONF_HOST in config:
-            influx_conf["host"] = config[CONF_HOST]
-
-        if CONF_PATH in config:
-            influx_conf["path"] = config[CONF_PATH]
-
-        if CONF_PORT in config:
-            influx_conf["port"] = config[CONF_PORT]
-
-        if CONF_SSL in config:
-            influx_conf["ssl"] = config[CONF_SSL]
-
-        queries = config[CONF_QUERIES]
+    try:
+        influx = get_influx_connection(conf, test_read=True)
+    except ConnectionError as exc:
+        _LOGGER.error(RETRY_MESSAGE, exc)
+        event_helper.call_later(
```

Comment:
Raise `PlatformNotReady` here.

Reply:
Oh ok.  Does that cause it to retry again later or does that just end setup until the user restarts HA?

Reply:
Oh I found it. So it tries again in a loop with an increasing delay each time.  That's great, makes sense, I'll use that instead. Can I also use that in `__init__.py` or does that not work for the main integration component and only for the platforms it provides?

Reply:
It only works for platforms.

Reply:
Ah bummer. Right I think I remember discussing that this option is available for newer config entries but not for legacy configs.  Ok thanks!